bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -1004,6 +1004,50 @@ impl Connection {
     }
 }
 
+#[cfg(feature = "unstable-serialize")]
+impl Connection {
+    /// Retrieves the size of the serialized connection
+    pub fn serialized_connection_length(&self) -> Result<usize, Error> {
```

Comment:
Is there a specific reason to use `usize` here? It seems easier to match the `C` api and use `u32`.

Reply:
I kind of thought it was more rusty to convert to usize. @camshaft thoughts?

Reply:
I'd use `usize` in any place where we're actually using a platform dependent size, (e.g. if we were using a `size_t`) but in this case it feels like a bit of a lie, since we know it will always be exactly a `u32`.

Also technically Rust (as a language) has support for [16 bit platforms](https://stackoverflow.com/questions/57618600/are-the-usize-isize-types-in-rust-guaranteed-to-always-be-either-32-or-64-bits), which means that converting the u32 to usize could panic on some super exotic platforms.

Reply:
The return value of this function will almost always be passed to alloc, which does use `usize`: https://doc.rust-lang.org/std/alloc/struct.Layout.html#method.from_size_align. So in general the `usize` value is the maximum addressable memory, in which case a u32 wouldn't actually be correct there.